packages/lib/src/utils/isEmpty.ts

```diff
@@ -1,9 +1,15 @@
-import _isEmpty from 'lodash/isEmpty';
-
 export const isEmpty = (value: unknown) => {
-  // short-circuit for performance
   if (value === undefined || value === null) {
     return true;
   }
-  return _isEmpty(value) as boolean;
+  if (typeof value === 'string') {
+    return value.length === 0;
+  }
+  if (value instanceof Map || value instanceof Set) {
+    return value.size === 0;
+  }
+  if (typeof value === 'object') {
+    return Object.keys(value).length === 0;
+  }
+  return false;
```

Comment:
value is array?
return Boolean(value)?

Reply:
Array is also an object, so `typeof [] === 'object'` and checking keys works for array as well. 

Also I dislike type coercion for following reasons:
1) it's slower then other stuff (well it's fast... but compared to other stuff). 
https://www.measurethat.net/Benchmarks/Show/11127/0/boolean-vs#latest_results_block
2) It's not explicit, you need to remember all the rules of converting one type to another. It's better to check against specific possible value as rule of thumb. ( !== null or !== undefined) Explicit is better then implicit, as is stated in many articles and books about programming, well unless it really hurts DX.